protocol/migrations/sqlite/1733428521_pinned_messages_add_on_delete_clause.up.sql

```diff
@@ -1,3 +1,13 @@
+-- Delete pinned messages that are not associated with any user message
+-- This makes sure the migration below works even if a message is deleted
+DELETE FROM pin_messages
```

Comment:
Hmm. So at the moment of running the migration, if we didn't yet receive the `message`, but we already received the `pin_message`, we will drop the information about the pinning action 🤔

Reply:
But this also means that further, if we receive the `pin_message` before the message itself (which is possible), we won't be able to save it into the `pin_message` table, because `message_id` foreign key will not be found. Sounds not good, but maybe (hopefully) I'm missing smth?

Reply:
Oh that's a great point, I didn't think about that! I guess we don't have a test covering that because it would have failed then.

I now realize that the original migration was already merged :grimacing: 

I wonder then if maybe I should revert it since there is that edge case you just found

Reply:
I wrote a test real quick that receives a Pin before the actual message and it fails in the query with `FOREIGN KEY constraint failed`, because like you said @igor-sirotin , the message is not found.

I'm not sure what's the best course of action. I can revert the migration or we can try to find a way to still save the pin when the message doesn't exist yet? cc @osmaczko since he proposed the `FOREIGN KEY` migration. Before that, I just had a query that deleted the pins when we deleted the message https://github.com/status-im/status-go/pull/6173#discussion_r1871961512

Reply:
I didn't think of it either, sorry for that. Removing `FOREIGN KEY` constrain seems like a way to go.

Reply:
Yeah removing sounds good to me 👍 
But why did we add `FOREIGN KEY` in the first place? To remove the rows in cascade? Then I guess we need a replacement for this behaviour?

Reply:
> But why did we add FOREIGN KEY in the first place? To remove the rows in cascade? Then I guess we need a replacement for this behaviour?

Yes you are right. I will add a new issue after this one to do the behaviour manually without needing the constraint. 
Similar to what I had here https://github.com/status-im/status-go/pull/6173#discussion_r1871961512

Reply:
Issue created here: https://github.com/status-im/status-go/issues/6246

@igor-sirotin @osmaczko can you give an approval to this one, I'll start working on the fix in  a few.